File: app/src/main/java/com/solt9029/editmasterandroid/service/ScoreService.java
package com.solt9029.editmasterandroid.service;

import com.solt9029.editmasterandroid.model.Score;

import java.util.List;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ScoreService {
    String BASE_URL = "http://editmasterapi.solt9029.com";

    @GET("scores/{id}")
    Single<Score> getScore(@Path("id") Integer id);

    @GET("scores/timeline")
    Single<List<Score>> getScoreTimeline(@Query("count") Integer count, @Query("keyword") String keyword, @Query("max_id") Integer maxId, @Query("since_id") Integer sinceId);
}

File: app/src/main/java/com/solt9029/editmasterandroid/service/ScoreService.kt
package com.solt9029.editmasterandroid.service

import com.solt9029.editmasterandroid.entity.Score
import io.reactivex.Single
import retrofit2.Response
import retrofit2.http.*

interface ScoreService {
    @GET("scores/{id}")
    fun getScore(@Path("id") id: Int?): Single<Score>

    @GET("scores/timeline")
    fun getScoreTimeline(@Query("count") count: Int?, @Query("keyword") keyword: String?,
                         @Query("max_id") maxId: Int?, @Query("since_id") sinceId: Int?): Single<List<Score>>

    @Headers("X-Requested-With:XMLHttpRequest")
    @POST("scores/create")
    fun createScore(@Body score: Score): Single<Response<Score>>

    companion object {
        const val BASE_URL = "http://editmasterapi.solt9029.com"
    }
}
